### PR TITLE
Scope: Stripe SEPA (European Union) now uses the Gateway API

### LINKED ISSUE #6207: Stripe SEPA (European Union) should use the Gateway API
## Acceptance Criteria

- [ ] Creates a payment using [SEPA Direct Debit (European Union)](https://stripe.com/docs/testing#sepa-direct-debit).

### PR DESCRIPTION
**Pull request by kjohnson**
Resolves #6207

Depends on #6213

## Description

This PR creates a new Stripe BECS Gateway using the Gateway API.

## Affects

This PR extracts the shared `canShowFields()` functionality into a shared sub-trait.

## Testing Instructions

SEPA payments require EUR-denominated bank account within the SEPA region.

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by ravinderk** [comment]
[@kjohnson](https://github.com/kjohnson) Is there any reason to make this pull request against `develop`.

**Comment by kjohnson** [comment]
@ravinderk per Jason's direction, instead of merging each gateway into an `epic` branch these `scope` PRs are independently deliverable and merge-able into `develop`.

**Comment by jonwaldstein** [comment]
@kjohnson this PR appears to be all the Stripe gateways - not just Stripe SEPA

**Comment by kjohnson** [comment]
@jonwaldstein each PR was built on top of the previous. So there is a dependency order.

**Comment by jonwaldstein** [comment]
@kjohnson ideally each completed gateway would have their own scope PR. This PR states its for Stripe SEPA but we would be merging all 4 gateways into develop with it which is confusing.  Honestly, we might be past the point of isolating each gateway into their own scope PR at this point.  It might be better to create an initial epic PR with the all ready approved Stripe gateways, then scope out the SEPA an BECS ones against develop so that they are truly isolated scopes.

**Comment by kjohnson** [comment]
@jonwaldstein that was my previous approach, but @JasonTheAdams requested that they be separate Scope PRs. They will just need to be merged in order and make sure that we don't merge one that includes another. I'm hoping that Gateways are an outlier.

**Comment by JasonTheAdams** [comment]
It's entirely possible we backed ourselves into a corner on this one. And that's ok; we're still all learning how to do this. If it makes the most sense to merge them into an Epic and shoot out a single zip, that's fine. The ideal, moving forward, is to build the Scope, stop, test it, merge it into `develop`, and then move onto the next scope.

**Comment by kjohnson** [comment]
@JasonTheAdams In this case the gateways were more interconnected than we had realized. They aren't really separate scopes, but together they make a very large scope. So there isn't an obvious solution.

**Comment by kjohnson** [comment]
@jonwaldstein if it helps, I can create a base branch and update the merge base so that it is easier to review.

**Comment by jonwaldstein** [comment]
@kjohnson that could work

File: src/PaymentGateways/Gateways/Stripe/BECSGateway.php [hunk 1]
```diff
@@ -19,6 +20,18 @@ class BECSGateway extends PaymentGateway
     use BECSMandateForm;
     use HandlePaymentIntentStatus;
 
+    /** @var array */
+    protected $errors = [];
```
Comment (jonwaldstein) [inline src/PaymentGateways/Gateways/Stripe/BECSGateway.php:24]: @kjohnson this should be `errorMessages` right?

Comment (kjohnson) [reply src/PaymentGateways/Gateways/Stripe/BECSGateway.php:24]: @jon correct. I can update the name.

Comment (kjohnson) [reply src/PaymentGateways/Gateways/Stripe/BECSGateway.php:24]: @jonwaldstein correct. I can update the name.

Comment (jonwaldstein) [reply src/PaymentGateways/Gateways/Stripe/BECSGateway.php:24]: okay cool

**Review by jonwaldstein — APPROVED** [review]
@kjohnson great job!

**Pull request merged**